include/pybind11/numpy.h

```diff
@@ -583,7 +597,9 @@ class array : public buffer {
 
         auto &api = detail::npy_api::get();
         auto tmp = reinterpret_steal<object>(api.PyArray_NewFromDescr_(
-            api.PyArray_Type_, descr.release().ptr(), (int) ndim, shape->data(), strides->data(),
+            api.PyArray_Type_, descr.release().ptr(), (int) ndim,
+            reinterpret_cast<Py_intptr_t*>(shape->data()),
```

Comment:
Could you add a comment somewhere that we know to remove it, once PyPy does the same thing a CPyhon?